samples/minimal/CMakeLists.txt

```diff
@@ -14,7 +14,7 @@
 #
 
 # Declare the minimum required CMake version
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.5)
```

Comment:
I think this needs to be documented somewhere.

Reply:
I'll see if it is documented anywhere now. It is mostly to get rid of the annoying warning everything you configure.
```cmake
Compatibility with CMake < 3.5 will be removed from a future version of CMake.
```

Another way might be:
```cmake
cmake_minimum_required(VERSION 2.8.12...3.28)
```

Though I added `IN_LIST` to the config file, which raises the minimum version to 3.3.

Reply:
Just to be clear: I don't mind increasing the version, I do believe that all platforms supported by current master should have (or have no trouble getting) CMake 3, but I just think it would be nice to document this requirement somewhere, probably in CMake build instructions.